handlebars/src/main/java/com/github/jknack/handlebars/internal/js/RhinoHandlebars.java

```diff
@@ -246,6 +169,39 @@ public CharSequence apply(final Object context, final Options options) throws IO
     });
   }
 
+  /**
+   * Serialize given object to JSON String.
+   *
+   * @param object to serialize
+   * @return JSON String
+   */
+  private static String toJson(final Object object) {
+    return new Gson().toJson(object);
+  }
+
+  /**
+   * Returns true for anything that we don' need to serialize as JSON.
+   *
+   * @param object to analyze
+   * @return whether object is Number, Boolean, String
+   */
+  private boolean isSimpleObject(final Object object) {
+
+    if (object instanceof Number) {
+      return true;
+    }
+    if (object instanceof Boolean) {
+      return true;
+    }
+    if (object instanceof CharSequence || object instanceof Character) {
+      return true;
+    }
+    if (object instanceof Scriptable) {
```

Comment:
Hm, I'd guess that GSON's serialization + deserialization results in something better than letting JS talk to the Java Date object directly, as that probably doesn't support [JS date object methods](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Date#Methods_2) . I didn't research it, though, whether it's really proper JS date object coming out of JSON serialization/deserialization with GSON.

Would be interesting to see what GSON assumes as timezone.